File: FrontEnd/Frontend_C#/SoftPac/SoftPacModel/CuentasAcreedorDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftPac.Model
{
    [Serializable]
    public class CuentasAcreedorDTO : CuentasBancariasDTO
    {
        private AcreedoresDTO acreedor;

        public CuentasAcreedorDTO() : base()
        {
            Acreedor = null;
        }

        public CuentasAcreedorDTO(int cuentaBancariaId, string tipoCuenta, string cci, bool activa,
                                  string numeroCuenta, MonedasDTO moneda, EntidadesBancariasDTO entidadBancaria,
                                  AcreedoresDTO acreedor)
            : base(cuentaBancariaId, tipoCuenta, cci, activa, numeroCuenta, moneda, entidadBancaria)
        {
            Acreedor = new AcreedoresDTO(acreedor);
        }

        public CuentasAcreedorDTO(CuentasAcreedorDTO other) : base(other)
        {
            this.Acreedor = new AcreedoresDTO(other.Acreedor);
        }

        public AcreedoresDTO Acreedor { get => acreedor; set => acreedor = value; }

    }
}

File: Backend_Java/SoftPacModel/src/main/java/pe/edu/pucp/softpac/model/CuentasAcreedorDTO.java
package pe.edu.pucp.softpac.model;

public class CuentasAcreedorDTO extends CuentasBancariasDTO {
    private AcreedoresDTO acreedor;
    public CuentasAcreedorDTO(){
        super();
        acreedor=null;
    }
    
    public CuentasAcreedorDTO(Integer cuenta_bancaria_id, String tipo_cuenta, String cci, Boolean activa, String numero_cuenta, MonedasDTO moneda, EntidadesBancariasDTO entidad_bancaria, AcreedoresDTO acreedor){
        super(cuenta_bancaria_id,tipo_cuenta,cci,activa,numero_cuenta,moneda,entidad_bancaria);
        this.acreedor=new AcreedoresDTO(acreedor);
    }
    
    public CuentasAcreedorDTO(CuentasAcreedorDTO cuenta_acreedor){
        super(cuenta_acreedor);
        this.acreedor=new AcreedoresDTO(cuenta_acreedor.getAcreedor());
    }

    /**
     * @return the acreedor
     */
    public AcreedoresDTO getAcreedor() {
        return acreedor;
    }

    /**
     * @param acreedor the acreedor to set
     */
    public void setAcreedor(AcreedoresDTO acreedor) {
        this.acreedor = acreedor;
    }

    
}
